nautobot/extras/tests/test_relationships.py

```diff
@@ -1363,3 +1364,80 @@ def required_relationships_test(self, interact_with="ui"):
+        with context_managers.web_request_context(self.user):
+            RelationshipAssociation.objects.create(
+                relationship=self.site_ip_relationship, source=self.site, destination=self.ip
+            )
+
+        with self.subTest("Source object change was generated"):
+            self.assertEqual(get_changes_for_model(self.site).count(), 1)
+            object_change = get_changes_for_model(self.site).first()
+            snapshots_added = object_change.get_snapshots()["differences"]["added"]
+            self.assertIn("relationships", snapshots_added)
+            self.assertEqual(
+                snapshots_added["relationships"]["sites-ipaddresses-m2m"]["destination"]["objects"][0]["id"],
+                str(self.ip.pk),
+            )
+
+        with self.subTest("Destination object change was generated"):
+            self.assertEqual(get_changes_for_model(self.ip).count(), 1)
+            object_change = get_changes_for_model(self.ip).first()
+            snapshots_added = object_change.get_snapshots()["differences"]["added"]
+            self.assertIn("relationships", snapshots_added)
+            self.assertEqual(
+                snapshots_added["relationships"]["sites-ipaddresses-m2m"]["source"]["objects"][0]["id"],
+                str(self.site.pk),
+            )
+
+    def test_change_logging_with_object_save(self):
+        """
+        Test that only a single objectchange is generated for associated objects when a RelationshipAssociation
+        is created and the associated object is saved in the same request
+        """
+        with context_managers.web_request_context(self.user):
+            self.ip.validated_save()
+            self.site.validated_save()
+            RelationshipAssociation.objects.create(
+                relationship=self.site_ip_relationship, source=self.site, destination=self.ip
+            )
+
+        with self.subTest("Source object change was generated"):
+            self.assertEqual(get_changes_for_model(self.site).count(), 1)
+            object_change = get_changes_for_model(self.site).first()
+            snapshots_added = object_change.get_snapshots()["differences"]["added"]
+            self.assertIn("relationships", snapshots_added)
+            self.assertEqual(
+                snapshots_added["relationships"]["sites-ipaddresses-m2m"]["destination"]["objects"][0]["id"],
+                str(self.ip.pk),
+            )
+
+        with self.subTest("Destination object change was generated"):
+            self.assertEqual(get_changes_for_model(self.ip).count(), 1)
+            object_change = get_changes_for_model(self.ip).first()
+            snapshots_added = object_change.get_snapshots()["differences"]["added"]
+            self.assertIn("relationships", snapshots_added)
+            self.assertEqual(
+                snapshots_added["relationships"]["sites-ipaddresses-m2m"]["source"]["objects"][0]["id"],
+                str(self.site.pk),
+            )
```

Comment:
Can/should we also test the case where an object is deleted, which should automatically delete the relationship-association, and make sure that change-logs as intended as well?